### PR TITLE
[FIX] initialize and postInitialize

### PR DESCRIPTION
**Pull request by vkalpias**
Call initialize and postInitialize on script instances in all cases regardless if they are enabled by enabling the Entity, the Script Component or the Script Instance in the right order

### CODE REVIEW

File: src/script/script.js [hunk 1]
```diff
@@ -536,6 +536,28 @@ pc.extend(pc, function () {
...
+                    if (this.initialize)
+                        this.entity.script._scriptMethod(this, pc.ScriptComponent.scriptMethods.initialize);
+                }
+
+                // post initialize script if not post initialized yet and still enabled
+                // (initilize might have disabled it so check this.enabled again)
+                // Warning: Do not do this if the script component is currently being enabled
+                // because in this case post initialize must be called after all the scripts
+                // in the script component have been initialized first
+                if (this._initialized && ! this._postInitialized && this.enabled && !this.entity.script._beingEnabled) {
```
Comment (H1Gdev) [inline src/script/script.js]: this._initialized is always true, so check this._initialized is not needed.

File: src/script/script.js [hunk 2]
```diff
@@ -536,6 +536,28 @@ pc.extend(pc, function () {
                     this.fire(this.enabled ? 'enable' : 'disable');
                     this.fire('state', this.enabled);
                 }
+
+                // initialize script if not initialized yet and script is enabled
+                if (! this._initialized && this.enabled) {
```
Comment (H1Gdev) [inline src/script/script.js]: I think that it is better to check `this.enable` only once.

```js
if (!this.enabled) return;
```
or
```js
if (this.enable) {
  if (! this._initialized) {
    ...
  }
  if (! this._postInitialized && !this.entity.script._beingEnabled) {
    ...
  }
}
```

Comment (vkalpias) [reply src/script/script.js]: The thing is that when you execute the initialize method then the script might no longer be enabled so we need to check `this.enabled` again.

File: src/script/script.js [hunk 3]
```diff
@@ -536,6 +536,28 @@ pc.extend(pc, function () {
...
+                    this._initialized = true;
+
+                    this.__initializeAttributes(true);
+
+                    if (this.initialize)
+                        this.entity.script._scriptMethod(this, pc.ScriptComponent.scriptMethods.initialize);
+                }
+
+                // post initialize script if not post initialized yet and still enabled
+                // (initilize might have disabled it so check this.enabled again)
```
Comment (H1Gdev) [inline src/script/script.js]: (initialize might have disabled ....)

File: src/script/script.js (hunk 2 above)
Comment (H1Gdev) [reply src/script/script.js]: Oh...  :disappointed:

Comment (H1Gdev) [reply src/script/script.js]: if this.enable is flip-flopped, I think that should call fire method after here.

**Pull request closed without merging**